src/DotNetty.Codecs.Redis/FixedRedisMessagePool.cs

```diff
@@ -74,12 +72,13 @@ static FixedRedisMessagePool()
                 stringToSimpleStringValues.Add(simpleString, redisMessage);
                 byteBufferToSimpleStringValues.Add(key, redisMessage);
             }
-            this.stringToSimpleStringMessages = new ReadOnlyDictionary<string, SimpleStringRedisMessage>(stringToSimpleStringValues);
-            this.byteBufferToSimpleStringMessages = new ReadOnlyDictionary<IByteBuffer, SimpleStringRedisMessage>(byteBufferToSimpleStringValues);
 
-            var errorToErrorValues = new Dictionary<string, ErrorRedisMessage>();
-            var byteBufferToErrorValues = new Dictionary<IByteBuffer, ErrorRedisMessage>();
-            foreach (string error in Errors)
+            this.stringToSimpleStringMessages = stringToSimpleStringValues.ToImmutableDictionary();
```

Comment:
why do you want immutable dictionary here? It's much slower than regular dictionary.

Reply:
It is a bit slower. I thought java version uses unmodifiable, let me change it back to plain dictionary.

Reply:
It was IReadOnlyDictionary in the initial port. Netty use plain HashMap. I will change it back to Dictionary.